src/routing/core/public_api.rs

```diff
@@ -145,10 +145,18 @@ impl Core {
         let target_name = msg.dst().name().ok_or(Error::CannotRoute)?;
         let dest_pk = *self.section_key_by_name(&target_name);
 
-        let targets: Vec<_> = targets
-            .into_iter()
-            .filter(|peer| self.msg_filter.filter_outgoing(msg, peer.name()).is_new())
-            .collect();
+        let mut targets = vec![];
+
+        for peer in presumed_targets {
```

Comment:
using filter_map will be better than this current block of code

Reply:
(using async, so cannot)